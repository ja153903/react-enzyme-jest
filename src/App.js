import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>
const Title = ({ text }) => <div>{text}</div>

export class App extends Component {
  state = {
    on: false,
    input: '',
    mainColor: 'blue',
    lifeCycle: '',
    hide: false
  }

  componentDidMount() {
    this.setState({
      lifeCycle: 'componentDidMount'
    })
  }

  componentWillReceiveProps() {
    this.setState({
      lifeCycle: 'componentWillReceiveProps'
    })
  }

  handleStrings(str) {
    return str === "Hello World" ? true : false;
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
          <h3 className={this.state.mainColor}>Everyone is welcome!</h3>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <Title text="jaime" />
        <ul></ul>
        <ul></ul>
        <ul></ul>
        <p className="button-state">{this.state.on ? "Yes!" : "No!"}</p>
        <button onClick={() => this.setState({ on: true})}>Click</button>
        <h2>{ this.state.input }</h2>
        <input type="text" 
        onChange={(e) => this.setState({ input: e.target.value })}/>
        <p className="lifeCycle">{this.state.lifeCycle}</p>
      </div>
    );
  }
}

//export default App;

export class Link extends Component {
  render() {
    return (this.props.hide ? null :
      <a href={this.props.address}>Click</a>)
    
  }
}
