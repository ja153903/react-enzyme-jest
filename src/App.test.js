import React from 'react';
import ReactDOM from 'react-dom';
import {App, Link} from './App';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

// need to do this for enzyme
configure({ adapter: new Adapter() });

// testing component did mount

// shallow rendering allows us to test the component as a unit
// without having to take care of its children
describe('<App /> shallow rendering', () => {
  it('should render App', () => {
    // context allows us to test with specific context in the component
    const wrapper = shallow(<App />, {
      context: {},
      disableLifecycleMethods: true
    })
    expect(wrapper.find("p").exists()).toBe(true)
    expect(wrapper.find("ul").length).toBe(3)
    expect(wrapper.find("p").length).toBe(2)
    expect(wrapper.find('[text="jaime"]').text()).toBe("<Title />")
    expect(wrapper.find({text: "jaime"}).exists()).toBe(true)
  })

  it('matches the snapshot', () => {
    const tree = shallow(<App />)
    expect(tree).toMatchSnapshot()
  })

  it('on button click changes p text', () => {
    const wrapper = shallow(<App />)
    const button = wrapper.find("button")
    expect(wrapper.find(".button-state").text()).toBe("No!")
    button.simulate('click') // targets prop based on the event
    expect(wrapper.find('.button-state').text()).toBe("Yes!")
  })
})

// you want to import js-dom installed to test the whole DOM

describe('<App /> mount rendering', () => {
  it('h1 contains the correct text', () => {
    // context allows us to test with specific context in the component
    // disableLifecycleMethods: true
    const wrapper = mount(<App />)
    
    // expect(wrapper.find("p").exists()).toBe(true)
    // expect(wrapper.find("ul").length).toBe(3)
    // expect(wrapper.find("p").length).toBe(1)
    // expect(wrapper.find('[text="jaime"]').text()).toBe("<Title />")
    // expect(wrapper.find({text: "jaime"}).exists()).toBe(true)
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
    wrapper.unmount();
  })

  it('matches the snapshot', () => {
    const tree = mount(<App />)
    expect(tree).toMatchSnapshot()
    tree.unmount();
  })
})

describe('<Link /> shallow rendering', () => {
  it('link component accepts address prop', () => {
    const wrapper = shallow(<Link address="www.google.com" />)
    expect(wrapper.instance().props.address).toBe("www.google.com")
  })

  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address='www.google.com' />)
    expect(wrapper.props().href).toBe("www.google.com")
  })

  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />)
    expect(wrapper.find('a').length).toBe(1)
    wrapper.setProps({ hide: true })
    // get returns the node at the given index of the current wrapper
    expect(wrapper.get(0)).toBeNull()
  })
})

// describe('<Link />', () => {
//   it('link component accepts address prop', () => {
//     const wrapper = shallow(<Link address="www.google.com" />)
//     expect(wrapper.instance().props.address).toBe("www.google.com")
//   })

//   it('a tag node renders href correctly', () => {
//     const wrapper = shallow(<Link address='www.google.com' />)
//     expect(wrapper.props().href).toBe("www.google.com")
//   })

//   it('returns null with true hide prop', () => {
//     const wrapper = shallow(<Link hide={false} />)
//     expect(wrapper.find('a').length).toBe(1)
//     wrapper.setProps({ hide: true })
//     // get returns the node at the given index of the current wrapper
//     expect(wrapper.get(0)).toBeNull()
//   })
// })

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
