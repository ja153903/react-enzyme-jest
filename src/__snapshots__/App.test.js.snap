// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
      <h3
        className="blue"
      >
        Everyone is welcome!
      </h3>
    </header>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Title
      text="jaime"
    >
      <div>
        jaime
      </div>
    </Title>
    <ul />
    <ul />
    <ul />
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <p
      className="lifeCycle"
    >
      componentDidMount
    </p>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is welcome!
    </h3>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <Title
    text="jaime"
  />
  <ul />
  <ul />
  <ul />
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
  <p
    className="lifeCycle"
  >
    componentDidMount
  </p>
</div>
`;
